### PR TITLE
derive(Arbitrary)

### LINKED ISSUE #4: Auto-deriving Arbitrary
Similar to what I did for quickcheck, it should be easy enough to write a derive macro for `Arbitrary`, perhaps even with some shrinking support for simple types. Should I do it in another crate or send a PR your way to integrate it?

### PR DESCRIPTION
**Pull request by llogiq**
This adds a small macro for deriving Arbitrary. No shrinking specialization yet. Any ideas on what to do for shrinking?

This closes #4

### CODE REVIEW

**Comment by nagisa** [comment]
NB: shrinking is not really implemented for the manual types either, I wouldn’t worry about that too much I think.

File: derive/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        s.gen_impl(quote! {
+            extern crate arbitrary;
+
+            use arbitrary::{Arbitrary, Unstructured};
+
+            gen impl Arbitrary for @Self {
+                fn arbitrary<U: Unstructured + ?Sized>(u: &mut U) -> Result<Self, U::Error> {
+
+
+                    Ok(match (u64::from(<u32 as Arbitrary>::arbitrary(u)?) * #count) >> 32 {
```
Comment (nagisa) [inline derive/src/lib.rs]: I’m having a hard time understanding what this means. We generate u32, convert it to u64, multiply by the number of variants and then drop the low bytes? What is the purpose of this complex logic? Why isn’t the genereated number taken modulo variant count?

Perhaps a comment would be appropriate.

Comment (llogiq) [reply derive/src/lib.rs]: This version introduces less bias than modulo and is due to [lemire](https://lemire.me/blog/2016/06/30/fast-random-shuffling/).

Comment (nagisa) [reply derive/src/lib.rs]: Please extend the comment with this blog post and also mention that this has a very slight bias.

**Comment by nagisa** [comment]
Thanks!

**Comment by llogiq** [comment]
You're welcome.

**Comment by nagisa** [comment]
I’ve took liberty of [publishing the crate](https://crates.io/crates/derive_arbitrary)

**Pull request merged**